fars/booking/bill.py

```diff
@@ -0,0 +1,48 @@
+import requests
+from getenv import env
+from django.utils.translation import gettext as _
+
+class BILLException(Exception):
+    pass
+
+class NotAllowedException(BILLException):
+    pass
+
+class BILLChecker:
+
+    def __init__(self):
+        self.api_url = env("BILL_API_URL")
+        self.user = env("BILL_API_USER")
+        self.password = env("BILL_API_PW")
+
+    def check_user_can_book(self, user, type, group=0):
+        group = 0 if group is None else group
+        try:
+            r = requests.get(self.api_url + "query?user=%s&group=%s&type=%s" % (user, group, type), auth=(self.user, self.password))
```

Comment:
Do we know if BILL is blocking as in the case of GeneriKey?

Reply:
Generikey isn't blocking per se, it's the requests library, as it waits for the response.
In this case, it doesn't matter because we have to wait for the response anyway. With Generikey we could just ignore the response because it's just a notification signal.